python/tests/test_ring.py

```diff
@@ -70,6 +73,28 @@ def test_harminv(self):
         self.assertAlmostEqual(ep, 11.559999999999999, places=places)
         self.assertAlmostEqual(fp, -0.08185972142450348, places=places)
 
+    def test_pade(self):
+        self.init()
+
+        self.sim.run(
+            mp.at_beginning(mp.output_epsilon),
+            self.p,
+            mp.after_sources(self.h),
+            until_after_sources=300,
+        )
+
+        freqs = np.linspace(
+            self.h.fcen - self.h.df / 2, self.h.fcen + self.h.df / 2, 1000
+        )
+
+        freq_domain = [self.p.freq_response(freq) for freq in freqs]
+
+        idx = find_peaks(
+            np.abs(freq_domain) ** 2 / max(np.abs(freq_domain) ** 2), prominence=1e-4
+        )[0]
+
+        self.assertAlmostEqual(freqs[idx[0]], self.h.modes[0].freq, places=3)
```

Comment:
Is it possible to test for multiple spectral peaks rather than just one?

Reply:
It is possible, but the `scipy.signal.find_peaks` algorithm I used requires some coaxing to only find the same modes that `Harminv` finds, so it feels a bit contrived. Is there a better way to identify peaks and compare them to the `Harminv` modes?

Reply:
Perhaps we just need to use a different ring resonator? For example, the ring resonator in this [tutorial](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#modes-of-a-ring-resonator) contains three closely spaced resonant modes.